### PR TITLE
NOISSUE - Add measuring time from pub to sub

### PR DESCRIPTION
**Pull request by mteodor**
### What does this do?
add measuring of time from publishing up to receiving a message in subscriber
synchronize publishers and subscribers on start end end of publishing
add a test message that can be normalized

### CODE REVIEW

File: tools/mqtt-bench/client.go [hunk 1]
```diff
@@ -127,30 +127,74 @@ func (c *Client) generate(ch chan *message, done chan bool) {
...
+	i := 0
 	token := (*c.mqttClient).Subscribe(c.MsgTopic, c.MsgQoS, func(cl mqtt.Client, msg mqtt.Message) {
-		mp := messagePayload{}
-		err := json.Unmarshal(msg.Payload(), &mp)
+
+		arrival := float64(time.Now().UnixNano())
+		mp, err := senml.Decode(msg.Payload(), senml.JSON)
 		if err != nil {
-			log.Printf("Client %s failed to decode message\n", clientID)
+			//log.Printf("Failed to decode message %s\n", err.Error())
```
Comment (anovakovic01) [inline tools/mqtt-bench/client.go]: Remove the code that is commented out.

File: tools/mqtt-bench/templates/config.toml [hunk 2]
```diff
@@ -1,25 +1,32 @@
...
 
 [test]
+<<<<<<< HEAD
 pubs = 100
 subs = 30
 count = 100
+=======
+pubs = 1 
+subs = 1
+>>>>>>> improve sync between pub and sub
```
Comment (anovakovic01) [inline tools/mqtt-bench/templates/config.toml]: You've committed conflict marks. Please resolve them.

File: tools/mqtt-bench/bench.go [hunk 3]
```diff
@@ -76,17 +83,17 @@ type Config struct {
...
 }
 
 // Benchmark - main benckhmarking function
 func Benchmark(cfg Config) {
 	var wg sync.WaitGroup
 	var err error
 
-	checkConnection(cfg.MQTT.Broker.URL, 1)
-	subTimes := make(SubTimes)
+	//checkConnection(cfg.MQTT.Broker.URL, 1)
```
Comment (anovakovic01) [inline tools/mqtt-bench/bench.go]: Remove the code that is commented out.

**Review by anovakovic01 — CHANGES_REQUESTED** [review]

**Review by dborovcanin — DISMISSED** [review]

File: tools/mqtt-bench/bench.go [hunk 4]
```diff
@@ -193,15 +216,121 @@ func Benchmark(cfg Config) {
...
+	if err != nil {
+		log.Fatalf("Cannot unmarshal payload")
+	}
+
+	msgByte, err := json.Marshal(sml)
+	if err != nil {
+		log.Fatalf("Failed to create test message")
+	}
+
+	// how many records to make message long sz bytes
```
Comment (drasko) [inline tools/mqtt-bench/bench.go]: Comments start with capital letters

File: tools/mqtt-bench/bench.go [hunk 5]
```diff
@@ -193,15 +216,121 @@ func Benchmark(cfg Config) {
...
+	n := (sz-len(tsByte))/len(msgByte) + 1
+	if sz < len(tsByte) {
+		n = 1
+	}
+
+	records := make([]senml.SenMLRecord, n)
+	records[0] = timeStamp
+	for i := 1; i < n; i++ {
+		// is this needed
```
Comment (drasko) [inline tools/mqtt-bench/bench.go]: Remove dead code

**Review by drasko — CHANGES_REQUESTED** [review]

**Review by drasko — APPROVED** [review]
LGTM

**Review by dborovcanin — APPROVED** [review]

**Review by anovakovic01 — APPROVED** [review]
LGTM!

**Pull request merged**